handlers/secrets_test.go

```diff
@@ -0,0 +1,236 @@
+		initialSecrets["without_label"] = genFakeSecret("without_label", "test", false)
+	}
+
+	return initialSecrets
+}
+
+type fakeDockerSecretAPIClient struct {
+	secrets map[string]swarm.Secret
+}
+
+func newFakeDockerSecretAPIClient() fakeDockerSecretAPIClient {
+	return fakeDockerSecretAPIClient{
+		secrets: getInitialSecrets(true),
+	}
+}
+
+func (c *fakeDockerSecretAPIClient) Reset() {
+	c.secrets = getInitialSecrets(true)
+}
+
+func (c *fakeDockerSecretAPIClient) SecretList(
+	_ context.Context,
+	options types.SecretListOptions,
+) ([]swarm.Secret, error) {
+	secrets := []swarm.Secret{}
+
+	for _, value := range c.secrets {
+		secrets = append(secrets, value)
+	}
+
+	return secrets, nil
+}
+
+func (c *fakeDockerSecretAPIClient) SecretCreate(
+	_ context.Context,
+	secretDesc swarm.SecretSpec,
+) (types.SecretCreateResponse, error) {
+	id := secretDesc.Name
+
+	newSecret := swarm.Secret{
+		ID: id,
+		Spec: swarm.SecretSpec{
+			Annotations: swarm.Annotations{
+				Name: secretDesc.Name,
+				Labels: map[string]string{
+					ownerLabel: ownerLabelValue,
+				},
+			},
+			Data: secretDesc.Data,
+		},
+	}
+
+	c.secrets[secretDesc.Name] = newSecret
+
+	return types.SecretCreateResponse{ID: id}, nil
+}
+
+func (c *fakeDockerSecretAPIClient) SecretRemove(_ context.Context, id string) error {
+	if _, ok := c.secrets[id]; !ok {
+		return fmt.Errorf("secret with id: %s not found", id)
+	}
+
+	delete(c.secrets, id)
+
+	return nil
+}
+
+func (c *fakeDockerSecretAPIClient) SecretInspectWithRaw(
+	_ context.Context,
+	name string,
+) (swarm.Secret, []byte, error) {
+	return swarm.Secret{}, nil, nil
+}
+
+func (c *fakeDockerSecretAPIClient) SecretUpdate(
+	_ context.Context,
+	id string,
+	version swarm.Version,
+	secret swarm.SecretSpec,
+) error {
+	return fmt.Errorf("returning error because it's not possible to update existing secrets with docker")
+}
+
+func Test_SecretsHandler(t *testing.T) {
+	dockerClient := newFakeDockerSecretAPIClient()
+	secretsHandler := MakeSecretsHandler(&dockerClient)
+
+	secretName := "testsecret"
+
+	t.Run("create managed secrets", func(t *testing.T) {
+		defer dockerClient.Reset()
+
+		secretValue := "testsecretvalue"
+		payload := fmt.Sprintf(`{"name": "%s", "value": "%s"}`, secretName, secretValue)
+		req := httptest.NewRequest("POST", "http://example.com/foo", strings.NewReader(payload))
+		w := httptest.NewRecorder()
+
+		secretsHandler(w, req)
+
+		resp := w.Result()
+
+		if resp.StatusCode != http.StatusCreated {
+			t.Errorf("expected status code '%d', got '%d'", http.StatusCreated, resp.StatusCode)
+		}
+
+		if _, secretExist := dockerClient.secrets[secretName]; !secretExist {
+			t.Errorf("secret `%s` was not created as expected", secretName)
+		}
+
+		if data := dockerClient.secrets[secretName].Spec.Data; !bytes.Equal(data, []byte(secretValue)) {
+			t.Errorf("want secret: `%s` to be equal `%s`, got: `%s`", secretName, secretValue, string(data))
+		}
+	})
+
+	t.Run("update managed secrets", func(t *testing.T) {
```

Comment:
it would be good to add some context to the test name, like `update managed secrets returns error` this will help future maintainers to understand what a test _should_ be doing

Reply:
Agreed

Reply:
Changed